LibreNMS/OS/Junos.php

```diff
@@ -59,7 +59,7 @@ public function discoverOS(Device $device): void
 
         preg_match('/Juniper Networks, Inc. (?<hardware>\S+) .* kernel JUNOS (?<version>[^, ]+)[, ]/', $device->sysDescr, $parsed);
         if (isset($data[2]['hrSWInstalledName'])) {
-            preg_match('/\[(.+)]/', $data[2]['hrSWInstalledName'], $parsedVersion);
```

Comment:
I think it’s better we do 
```suggestion
            preg_match('/\[([^]]+)/', $data[2]['hrSWInstalledName'], $parsedVersion);
```

Reply:
We need to ignore things that do not have JUNOS in the package name because based on observational data non JUNOS packages can exist with different version numbers.     I'm not sure why we just 'assume' that the second package installed will have the correct version number on it ...    Perhaps 'most of the time' it does.

I didn't want to go too far down this rabbit hole, just fill it in around the sides a bit for known rabbits of a particular size ... ??

Reply:
He meant it is better to greedy match everything excluding ].

But both ways work imo, and we have test data to confirm.